### PR TITLE
Add the execution permission to the scripts for the runAsMe mode

### PR DESCRIPTION
**Pull request by aminelouati**

### CODE REVIEW

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/ClasspathUtils.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                }
+                parentFile = parentFile.getParentFile();
+            }
+            return new File(".").getAbsolutePath();
+        } catch (Exception e) {
+            return new File(".").getAbsolutePath();
+        }
+    }
+
+    public static String findSchedulerHome() {
```
Comment (Mykhailenko) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/ClasspathUtils.java]: Maybe it makes sense to make this method `synchronized`?

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java [hunk 2]
```diff
@@ -126,4 +129,19 @@ private OSProcessBuilder getOsProcessBuilder(TaskContext context, File workingDi
         }
         return processBuilder;
     }
+
+    private void addExecutionPermissionToScripts() throws IOException {
+        String schedulerHome = ClasspathUtils.findSchedulerHome();
```
Comment (nebilbenmabrouk) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java]: Include this instruction in the if block
String schedulerHome = ClasspathUtils.findSchedulerHome();

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java [hunk 3]
```diff
@@ -126,4 +129,19 @@ private OSProcessBuilder getOsProcessBuilder(TaskContext context, File workingDi
...
+        //Add execution permissions to scripts except for windows OS
+        if (!StandardSystemProperty.OS_NAME.value().toLowerCase().contains("windows")) {
+            File scriptsDirectory = new File(schedulerHome, "dist/scripts/processbuilder/linux");
+            if (scriptsDirectory != null) {
+                for (File script : scriptsDirectory.listFiles()) {
+                    script.setExecutable(true, false);
```
Comment (nebilbenmabrouk) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java]: Please, change this instruction to add execution rights to all users:
script.setExecutable(true, false);

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java (hunk 2 above)
Comment (aminelouati) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java]: ok

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java (hunk 3 above)
Comment (aminelouati) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedProcessBuilderCreator.java]: it is the case with script.setExecutable(true, false);
The second parameter is ownerOnly

**Pull request merged**